src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1926,4 +1942,27 @@ public static String getVersionName() {
         // value
         return "1.1.12";
     }
+
+    /**
+     * A {@link Future}  whose result can be set by a {@link #set(Object)} or {@link #setException(Throwable)}
+     */
+    final static class SettableFuture<V> extends FutureTask<V> {
```

Comment:
@RandalliLama There is another option - we inherit from the ThreadPoolExecutor and override `afterExecute` inside we check if "task" is a `Future` and check it's result. If result is `RuntimeException` we crash.